File: SortingInCSharp/SortingInCSharp/MergeSort.cs
using System;
namespace SortingInCSharp
{
	public static class MergeSort
	{
		public static int[] SortAscending(int[] array)
		{
			int len = array.Length;
			Divide(array, 0, len-1);
			return array;
		}

		public static void Divide(int[] array, int left, int right)
		{
			if(left >= right)
			{
				return;
			}

			int mid = left + (right - left) / 2;

            Divide(array, left, mid);
			Divide(array, mid + 1, right);
			Conquer(array, left, mid, right);

		}

		public static void Conquer(int[] array, int left, int mid, int right)
		{
			int lengthArrayLeft = mid - left + 1;
			int lengthArrayRight = right - mid;
			int[] leftArray = new int[lengthArrayLeft];
			int[] rightArray = new int[lengthArrayRight];
			int i, j;

			for(i = 0;i < lengthArrayLeft; i++)
			{
				leftArray[i] = array[left + i];
			}

			for(i = 0; i< lengthArrayRight; i++)
			{
				rightArray[i] = array[mid + 1 + i];
			}

			i = 0;
			j = 0;
			int k = left;
			while(i < lengthArrayLeft && j < lengthArrayRight)
			{
				if (leftArray[i] < rightArray[j])
				{
					array[k++] = leftArray[i++];
				}
				else
				{
					array[k++] = rightArray[j++];
				}
			}

			while (i < lengthArrayLeft)
			{
				array[k++] = leftArray[i++];
			}

            while (j < lengthArrayRight)
            {
                array[k++] = rightArray[j++];
            }
        }
	}
}

File: SortingInPython/MergeSort.py

def mergeSort(array):
    l = len(array)

    divide(array, 0, l-1)



def divide(array, left, right):

    if left >= right:
        return
    
    mid = int(left + (right-left)/2)

    divide(array, left, mid)
    divide(array, mid+1, right)
    conquer(array, left, mid, right)



def conquer(array, left, mid, right):

    leftArrayLength = mid-left+1
    rightArrayLength = right-mid

    leftArray = [None] * leftArrayLength
    rightArray = [None] * rightArrayLength

    i=j=0

    for i in range(leftArrayLength):
        leftArray[i] = array[left + i]


    for i in range(rightArrayLength):
        rightArray[i] = array[mid + 1+ i]
    
    i = j = 0
    k = left

    while i < leftArrayLength and j < rightArrayLength:
        if leftArray[i] < rightArray[j]:
            array[k] = leftArray[i]
            i = i+1

        else:
            array[k] = rightArray[j]
            j = j+1
        
        k = k+1

    
    while i < leftArrayLength:
        array[k] = leftArray[i]
        k = k+1
        i = i+1

    while j < rightArrayLength:
        array[k] = rightArray[j]
        k = k+1
        j = j+1


def display(array):
    l = len(array)
    for x in range(l):
        print(array[x])





array = [-2, 45, -12, 11, -9 ]
mergeSort(array)

display(array)